### ISSUE TITLE
Won't Run From Subdirectory

### ISSUE THREAD

**Issue by Derjyn** [eveseat/seat#407]
Nginx environment. Everything works, except that I want to be able to run SEaT from http://domain.com/seat.

How to achieve this?

.env file has:
APP_URL=http://domain.com/seat

/etc/nginx/sites-available/seat has:
server_name domain.com/seat;

Also with the /etc/nginx/sites-available/seat, tried commenting out server_name, and also just setting it to domain.com

What am I missing? Figure this would be easier...

**Comment by TyrHeimdal** [comment]
You cannot set the subdirectory in the server_name - it is only for the virtual host (domain.com). Use root to set this. See: http://nginx.org/en/docs/beginners_guide.html for how this works. You want to place it in your *existing* virtual host (assuming you already have a site under domain.com).

For example:
`server {
...
    location / {
        root /var/www;
    }

    location /seat/ {
        root /path/to/seat;
    }
...
}`

**Comment by Derjyn** [comment]
Well, I have the following in my default:

```
server {
    listen 80 default_server;
    listen [::]:80 default_server;

    server_name test.com www.test.com;

    location / {
        root /var/www/html;        
        index index.html index.htm;
        try_files $uri $uri/ =404;
    }

    location /seat/ {
        root /var/www/seat/public/;
        try_files \$uri \$uri/ /index.php?\$query_string;
    }

    location ~ /seat/.+\.(jpg|jpeg|gif|css|png|js|ico|xml)$ {
        root /var/www/seat/public/;
    }

    location ~ /seat/.+\.php$ {
       try_files \$uri /index.php =404;
       fastcgi_pass unix:/run/php/php7.1-fpm.sock;
       fastcgi_param SCRIPT_FILENAME \$document_root\$fastcgi_script_name;
       include fastcgi_params;
    }
}
```
The base site "test.com" (just a basic html file) works fine.  However, "test.com/seat" returns a 404. If I follow the SEaT installation instructions to a t, and use the seat nginx virtual host file and such, SEaT takes over and then "test.com" goes right to SEaT.  Obviously, this is not wanted, as I have other subdirectories with apps and such I want to point to, as well as the main site. I just want to be able to type in "test.com/site" and have it go to site, and type in "test.com" and have it go to the main site, and so on.

Doing the simple addition to the virtual host file, with the snippet you provided, also presents a 404. I know SEaT is installed and running- doing things following the install instructions works fine (minus the hijacking of the site). Able to generate an admin login URL, get to the dashboard, sync up and login via ESI/SSO and all that.

**Comment by TyrHeimdal** [comment]
Hi,
- Do not escape the variables starting with $ (remove the backslashes)
- Is your PHP-FPM socket located at /run/php/php7.1-fpm.sock?
- Does the PHP-FPM pool user/group have access to your /var/www/seat/public/ directory (and at a minimum execution access to all parent directories)?

These are the most obvious causes of why it might fail.

**Comment by Derjyn** [comment]
Hello,

I've since updated the snippet, removing the slashes (though this was from official PHP docs somewhere, don't have the link handy).  PHP-FPM and the user/group access was fine.

At any rate, we ended up making a sub-domain and updating the config to reflect that.  Things worked fine then.  Still, it would be nice to be able to utilize logical paths without running into a wall!

Thanks for your assistance.
